util/reggen/access.py

```diff
@@ -90,7 +90,7 @@ def dv_rights(self) -> str:
         '''Return a UVM access string as used by uvm_field::set_access().'''
         if self.key == 'r0w1c':
             return 'W1C'
-        return self.value[1].name
+        return str(self.value[1])
```

Comment:
it looks like this causes issues in the generated UVM classes, since `str(self.value[1])` also returns the name of the enum itself (e.g. `SwAccess.W1C`), instead of just the name of the enum value (e.g. `W1C`):
```
UVM_ERROR @         0 ps: (uvm_reg_field.svh:836) [RegModel] Access policy 'SwAccess.W1C' for field 'alert_handler_reg_block.intr_state.classa' is not defined. Setting to RW
UVM_ERROR @         0 ps: (uvm_reg_field.svh:836) [RegModel] Access policy 'SwAccess.W1C' for field 'alert_handler_reg_block.intr_state.classb' is not defined. Setting to RW
UVM_ERROR @         0 ps: (uvm_reg_field.svh:836) [RegModel] Access policy 'SwAccess.W1C' for field 'alert_handler_reg_block.intr_state.classc' is not defined. Setting to RW
UVM_ERROR @         0 ps: (uvm_reg_field.svh:836) [RegModel] Access policy 'SwAccess.W1C' for field 'alert_handler_reg_block.intr_state.classd' is not defined. Setting to RW
UVM_ERROR @         0 ps: (uvm_reg_field.svh:836) [RegModel] Access policy 'SwAccess.RW' for field 'alert_handler_reg_block.intr_enable.classa' is not defined. Setting to RW

```
maybe `str(str(self.value[1].name)` works?

Reply:
Huh, that's odd. Reading it again, I think the fix is even simpler: change the declared type of the `dv_rights` function to be `SwAccess`. I've tried it locally and things look like they work. Will re-push with that version.